Dockerfile

```diff
@@ -1,74 +1,103 @@
-            ln -s /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/webstorm.sh /opt/run-ide.sh \
-            ;; \
-    esac && \
-    mkdir -p ${BASE_MOUNT_FOLDER}/${PRODUCT_NAME} && \
-    mkdir /etc/default/jetbrains && \
-    for f in "${BASE_MOUNT_FOLDER}" "/${PRODUCT_NAME}-${PRODUCT_VERSION}" "/etc/passwd" "/etc/default/jetbrains"; do \
-      echo "Changing permissions on ${f}" && chgrp -R 0 ${f} && \
-      chmod -R g+rwX ${f}; \
-    done && \
-    echo "idea.config.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/config" > /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.system.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/caches" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.plugins.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/plugins" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties && \
-    echo "idea.log.path=${BASE_MOUNT_FOLDER}/${PRODUCT_NAME}/logs" >> /${PRODUCT_NAME}-${PRODUCT_VERSION}/bin/idea.properties
+#
+# Copyright 2019-2020 JetBrains s.r.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
-# Copy fluxbox configuration
-COPY --chown=0:0 config/fluxbox /home/user/.fluxbox/init
 
-# Copy predefined configs
-COPY --chown=0:0 config/etc /etc/
+# Stage 1. Prepare JetBrains IDE with Projector.
+#   1. Downloads JetBrains IDE packaging by given downloadUrl build argument.
+#   2. If buildGradle build argument is set to false, then consumes built Projector assembly from the host.
+#       2.1 Otherwise starts Gradle build of Projector Server and Projector Client.
+#   3. Copies static files to the Projector assembly (entrypoint, launcher, configuration).
+FROM registry.access.redhat.com/ubi8-minimal:8.3-298 as projectorAssembly
+ENV PROJECTOR_DIR /projector
+ENV JAVA_HOME /usr/lib/jvm/java-11
+ARG downloadUrl
+ARG buildGradle
+ADD jetbrains-editor-images/static $PROJECTOR_DIR
+ADD projector-client $PROJECTOR_DIR/projector-client
+ADD projector-server $PROJECTOR_DIR/projector-server
+WORKDIR /downloads
+RUN microdnf install -y wget findutils tar gzip unzip java-11-openjdk-devel \
+    && wget -q $downloadUrl -O - | tar -xz \
+    && find . -maxdepth 1 -type d -name * -exec mv {} $PROJECTOR_DIR/ide \;
+WORKDIR $PROJECTOR_DIR/projector-server
+RUN if [ "$buildGradle" = "true" ]; then ./gradlew clean; else echo "Skipping gradle build"; fi \
```

Comment:
did you forget to install (download, unpack) gradle? Or did it get added to the `java-11-openjdk-devel` rpm bundle?

Reply:
No, everything is correct here, there is a Gradle Wrapper inside Projector Server and Projector Client. It is enough to have only `JAVA_HOME` environment variable to build the project. Gradle Wrapper does the all stuff for build.